cartoframes/client/sql_client.py

```diff
@@ -0,0 +1,208 @@
+        status and information of the job."""
+        try:
+            return self._client.execute_long_running_query(query)
+        except CartoException as e:
+            print('Error: {}'.format(e))
+
+    def distinct(self, table_name, column_name):
+        """Get the distict values and their count in a table
+        for a specific column."""
+        query = '''
+            SELECT {0}, COUNT(*) FROM {1}
+            GROUP BY 1 ORDER BY 2 DESC
+        '''.format(column_name, table_name)
+        output = self.query(query)
+        return list(map(lambda x: (x.get(column_name), x.get('count')), output))
+
+    def count(self, table_name):
+        """Get the number of elements of a table."""
+        query = 'SELECT COUNT(*) FROM {};'.format(table_name)
+        output = self.query(query)
+        return output[0].get('count')
+
+    def bounds(self, query):
+        """Get the bounds of the geometries in a table."""
+        output = self.query('''
+            SELECT ARRAY[
+                ARRAY[st_xmin(geom_env), st_ymin(geom_env)],
+                ARRAY[st_xmax(geom_env), st_ymax(geom_env)]
+            ] bounds FROM (
+                SELECT ST_Extent(the_geom) geom_env
+                FROM ({}) q
+            ) q;
+        '''.format(query))
+        return output[0].get('bounds')
+
+    def schema(self, table_name, raw=False):
+        """Show information about the schema of a table.
+        Setting raw=True is returns a JSON with the data."""
+        query = 'SELECT * FROM {0} LIMIT 0'.format(table_name)
+        output = self.query(query, verbose=True)
+        fields = output.get('fields')
+        rows = []
+        output = {}
+        for key in fields:
+            field = fields.get(key)
+            value = field.get('type')
+            output[key] = value
+            row = [key, value]
+            rows.append(row)
+        if raw:
+            return output
+        else:
+            self._print_table(rows, columns=['Column name', 'Column type'], padding=[10, 5])
+        
+    def describe(self, table_name, column_name):
+        """Show information about a column in a specific table."""
+        column_type = self._get_column_type(table_name, column_name)
+        stats = ['COUNT(*)']
+        if column_type == 'number':
+            stats.append('AVG({})'.format(column_name))
+            stats.append('MIN({})'.format(column_name))
+            stats.append('MAX({})'.format(column_name))
+        query = '''
+            SELECT {0}
+            FROM {1}
+        '''.format(','.join(stats), table_name)
+        output = self.query(query, verbose=True)
+        fields = output.get('rows')[0]
+        rows = []
+        for key in fields:
+            value = fields.get(key)
+            row = [key, round(value, 2)]
+            rows.append(row)
+        self._print_table(rows, padding=[5, 10])
+        print('type: {}'.format(column_type))
+
+    def create_table(self, table_name, columns, cartodbfy=True):
```

Comment:
I like the transparency in building the query as applied here: https://github.com/CartoDB/cartoframes/blob/develop/cartoframes/data/dataset.py#L535

Reply:
And AFAIK concatenating strings with += is slower than templating with format